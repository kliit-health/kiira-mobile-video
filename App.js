import React, {PureComponent} from 'react';
import {
  View,
  Alert,
  BackHandler,
  AppState,
  LogBox,
  Platform,
} from 'react-native';
import {connect} from 'react-redux';
import AppNavigator from './src/navigator';
import {showOrHideModal} from './src/components/customModal/action';
import CustomLoader from './src/components/customLoader';
import CustomModal from './src/components/customModal';
import CustomToast from './src/components/customToast';
import {getBottomSpace} from './src/components/iPhoneXHelper';
import firebase from 'react-native-firebase';
import {
  setFcmToken,
  setAppState,
  setAppScreen,
} from './src/screens/authLoading/action';
import {signOut} from './src/screens/account/action';
import Constant from './src/utils/constants';
import FastImage from 'react-native-fast-image';
// import BackgroundTask from 'react-native-background-task';
// import BackgroundTimerMain from 'react-native-background-timer';
import {updateStatus} from './src/utils/firebase';
import {NavigationService} from './src/navigator';
import {setCurrentRoute, setPreviousRoute} from './src/redux/actions';

class App extends PureComponent {
  constructor(props) {
    super(props);
    firebase.app();
    this.state = {
      appState: AppState.currentState,
    };
  }

  async componentDidMount() {
    LogBox.ignoreAllLogs();
    BackHandler.addEventListener(
      'hardwareBackPress',
      this.handleBackButtonClick,
    );
    // BackgroundTask.schedule();
    AppState.addEventListener('change', this._handleAppStateChange);
    FastImage.preload([
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F158453698551228C511E5-726E-4A6A-B48F-5789FB54A554.jpg?alt=media&token=0e896a21-1c3e-4fb8-b401-a3724d60339b',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1585955584718B9108868-3D87-4332-997C-D57BDC94A9CE.jpg?alt=media&token=fd676602-f4d7-41de-bc31-d37ca8e10d94',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F158278580245518424.jpeg?alt=media&token=cb39b2f9-9712-4307-9521-fc346e8d9120',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F158275803271418421.jpeg?alt=media&token=ec94cd04-0010-4631-96f0-ea698a1aad96',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1580857797665E118A4E7-CBC3-4337-9397-631C8FC63164.jpg?alt=media&token=14f0e805-28b0-4dfe-9cc4-820344d0a465',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F158949711060350622E2C-7688-4C0D-98E6-8E3F1D22CF45.jpg?alt=media&token=ff56a38b-abdf-4774-9eb6-7c380a7ca091',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1591229415428C4A07D08-AB4C-48DF-959E-45E28EA2D5F4.jpg?alt=media&token=e494faff-8eb5-4e8e-adce-21cf113597fd',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1591206219146F08CEA92-EDB4-4DEC-9D5A-1F9828C37BBD.jpg?alt=media&token=21cee628-6a1a-44f8-bdce-cd6bfacc91f1',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F159122772273362D509B0-AD92-49C2-AB0F-9B390AEB7DC9.jpg?alt=media&token=4997aa43-9f60-46aa-96ce-df08ffff8b8e',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1583730173970C2B96E94-916B-445A-999C-8E0B1583B10C.jpg?alt=media&token=3c6c68dd-b22d-4f51-aebb-1f0783895cae',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1588710676616A35BA904-2E4E-47B8-9A0F-F8A43009F2D5.jpg?alt=media&token=dd075403-d180-4ad3-9e26-7a468b258434',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1583348784419863E37D8-BCB6-443B-A3CF-381194EFBF02.jpg?alt=media&token=cccf1b7a-2b90-43fc-8827-9eb900f90af1',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F1589390627635F1D06990-A2A5-4D54-BAD5-990CFA681E74.jpg?alt=media&token=fde22ced-bb50-46a3-8d92-6e3c41f72bd7',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kliit-health-app.appspot.com/o/Kliit%2F158991152208724457F6C-FD35-4484-89B4-B1FFD8CD61FB.jpg?alt=media&token=8a1a2096-a8a8-445f-bb88-c473814331ba',
      },
      {
        uri:
          'https://firebasestorage.googleapis.com/v0/b/kiira-health-app.appspot.com/o/Kiira%2Fplaceholder.png?alt=media&token=ea401fa3-3f5c-4c29-9109-f1d0e4bbffdf',
      },
    ]);
  }

  componentDidUpdate() {
    const {isActive, userData} = this.props;
    if (isActive && userData && userData.uid && !userData.isOnline) {
      const updateStatusParams = {
        uid: userData.uid,
        updatedData: {
          isOnline: isActive,
        },
      };
      updateStatus(updateStatusParams);
    } else if (!isActive && userData && userData.uid) {
      const updateStatusParams = {
        uid: userData.uid,
        updatedData: {
          isOnline: isActive,
        },
      };
      updateStatus(updateStatusParams);
    }
  }

  componentWillUnmount() {
    // firebase.auth().signOut();
    BackHandler.removeEventListener(
      'hardwareBackPress',
      this.handleBackButtonClick,
    );
  }

  _handleAppStateChange = (nextAppState) => {
    const {setState, signOut} = this.props;

    if (nextAppState !== 'active') {
      // BackgroundTask.define(() => {
      //   setState(false);
      //   const payload = {
      //     navigation: this.navigator._navigation,
      //     isLoaderShow: false,
      //   };
      //   signOut(payload);
      //   Alert.alert(
      //     'Log Out',
      //     'For your security, you have been logged out.',
      //     [{text: 'OK', onPress: () => {}}],
      //     {cancelable: false},
      //   );
      //   BackgroundTask.finish();
      // });
    }
  };

  handleBackButtonClick() {
    setTimeout(() => {
      Alert.alert(
        'Exit App',
        'Are you sure you want to exit the App?',
        [
          {
            text: 'OK',
            onPress: () => {
              BackHandler.exitApp();
            },
          },
          {
            text: 'Cancel',
            onPress: () => {
              console.log('Cancel Pressed');
            },
          },
        ],
        {cancelable: false},
      );
    }, 400);
    return true;
  }

  getCurrentRouteName(navigationState) {
    if (!navigationState) {
      return null;
    }
    const route = navigationState.routes[navigationState.index];
    if (route.routes) {
      return this.getCurrentRouteName(route);
    }
    return route.routeName;
  }

  render() {
    const {
      spinnerState,
      errorMessage,
      hideErrorModal,
      showModalError,
      toastState,
      setScreen,
      setCurrentRoute,
      setPreviousRoute,
    } = this.props;

    return (
      <View
        style={{
          flex: 1,
          marginBottom: getBottomSpace(),
        }}>
        <AppNavigator
          ref={(nav) => {
            this.navigator = nav;
            NavigationService.navigator = nav;
          }}
          onNavigationStateChange={(prevState, currentState) => {
            const currentScreen = this.getCurrentRouteName(currentState);
            const prevScreen = this.getCurrentRouteName(prevState);
            const obj = {
              currentScreen,
              prevScreen,
            };
            setScreen(obj);
            setCurrentRoute(currentScreen);
            setPreviousRoute(prevScreen);
          }}
        />
        {showModalError ? (
          <CustomModal
            onPressErrorButtonOk={() => hideErrorModal()}
            showLoader={showModalError}
            errorMsg={errorMessage}
          />
        ) : null}

        {spinnerState.showLoader ? (
          <CustomLoader
            showLoader={spinnerState.showLoader}
            textMsg={spinnerState.textMessage}
          />
        ) : null}

        {toastState.showToast ? (
          <CustomToast
            showToast={toastState.showToast}
            textMsg={toastState.textMessage}
            dispatch={toastState.dispatch}
            delay={toastState.delay}
          />
        ) : null}
      </View>
    );
  }
}

const mapStateToProps = (state) => ({
  spinnerState: state.loaderReducer,
  toastState: state.toastReducer,
  showModalError: state.modalReducer.showModalError,
  errorMessage: state.modalReducer.errorMessage,
  userData: state.authLoadingReducer.userData,
  isActive: state.authLoadingReducer.isActive,
});

const mapDispatchToProps = (dispatch) => ({
  hideErrorModal: () => dispatch(showOrHideModal()),
  setToken: (value) => dispatch(setFcmToken(value)),
  setState: (value) => dispatch(setAppState(value)),
  signOut: (value) => dispatch(signOut(value)),
  setScreen: (value) => dispatch(setAppScreen(value)),
  setCurrentRoute: (value) => dispatch(setCurrentRoute(value)),
  setPreviousRoute: (value) => dispatch(setPreviousRoute(value)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
