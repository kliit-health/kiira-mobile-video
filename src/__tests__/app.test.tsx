import 'react-native';
import renderer from 'react-test-renderer';
import App from '../../App';
import React from 'react';
import BillingAndInsurance from '../screens/common/billingAndInsurance'; 
import EmergencyContact from '../screens/common/emergencyContact';
import Pharmacy from '../screens/common/pharmacy';

test('renders correctly', () => {
    const tree = renderer.create(<App />);
    expect(tree).toBeDefined();    
});  
 
describe('Testing my BillingAndInsurance', () => { 
    const billTree = renderer.create(<BillingAndInsurance navigation/>);
    test('billTree', () => {
        expect(billTree).toBeDefined();
    });

    const toggleModal = require('../screens/common/billingAndInsurance');  
    test('toggleModal', () => {
        expect(toggleModal(null, "insurance", "member")).toBeDefined();
    });

    const canConfirm = require('../screens/common/billingAndInsurance/editModal'); 
    test('canConfirm', () => {
        expect(canConfirm('first&last', 'last', '333-333-3333', '333-333-3333', 'Mother')).toBeTruthy();
        expect(canConfirm('&last', 'last', '333-333-3333', '333-333-3333', 'Mother')).toBeFalsy();
        expect(canConfirm('first&', 'last', '333-333-3333', '333-333-3333', 'Mother')).toBeFalsy();
        expect(canConfirm('first', 'last', '333-333-333', '333-33-3333', 'Mother')).toBeFalsy();
        expect(canConfirm('first', 'last', '333-333-333', '333-33-3333', '')).toBeFalsy();
    });
});

describe('Testing my EmergencyContact', () => {  
    const emergencyTree = renderer.create(<EmergencyContact navigation/>);
    test('emergencyTree', () => {
        expect(emergencyTree).toBeDefined();
    });

    const toggleModal = require('../screens/common/emergencyContact');   
    const emergencyContactInfo= {
        firstName: "firstName",
        lastName: "lastName",
        phoneNumber: "333-333-3333",
        secondNumber: "444-444-4444",
        relationship: "Mother",
    };
    test('toggleModal', () => {
        expect(toggleModal(null, emergencyContactInfo)).toBeDefined();
    });

    const phoneNumberValidation = require('../screens/common/emergencyContact'); 
    test('phoneNumberValidation', () => {
        expect(phoneNumberValidation('333-333-3333')).toBeTruthy();
        expect(phoneNumberValidation('33d-333-3333')).toBeFalsy();
        expect(phoneNumberValidation('333-333-333')).toBeFalsy();
        expect(phoneNumberValidation('333-333-333d')).toBeFalsy();
        expect(phoneNumberValidation('3333333333')).toBeFalsy();
    });

    const setValidPhoneNumber = require('../screens/common/emergencyContact'); 
    test('setValidPhoneNumber', () => {
        expect(setValidPhoneNumber('333-333-3333', true)).toBeTruthy();
        expect(setValidPhoneNumber('333-333-3333', false)).toBeTruthy();
        expect(setValidPhoneNumber('33d-333-3333', true)).toBeFalsy();
        expect(setValidPhoneNumber('33d-333-3333', false)).toBeFalsy();
        expect(setValidPhoneNumber('333-333-333', true)).toBeFalsy();
        expect(setValidPhoneNumber('333-333-333', false)).toBeFalsy();
        expect(setValidPhoneNumber('333-333-333d', true)).toBeFalsy();
        expect(setValidPhoneNumber('333-333-333d', false)).toBeFalsy();
        expect(setValidPhoneNumber('3333333333', true)).toBeFalsy();
        expect(setValidPhoneNumber('3333333333', false)).toBeFalsy();
    }); 

    const canConfirm = require('../screens/common/emergencyContact/editModal'); 
    test('canConfirm', () => {
        expect(canConfirm('first&last', 'last', '333-333-3333', '333-333-3333', 'Mother')).toBeTruthy();
        expect(canConfirm('&last', 'last', '333-333-3333', '333-333-3333', 'Mother')).toBeFalsy();
        expect(canConfirm('first&', 'last', '333-333-3333', '333-333-3333', 'Mother')).toBeFalsy();
        expect(canConfirm('first', 'last', '333-333-333', '333-33-3333', 'Mother')).toBeFalsy();
        expect(canConfirm('first', 'last', '333-333-333', '333-33-3333', '')).toBeFalsy();
    });
});

describe('Testing my Pharmacy', () => {  
    const pharmacyTree = renderer.create(<Pharmacy navigation/>);
    test('pharmacyTree', () => {
        expect(pharmacyTree).toBeDefined();
    });

    const handleOnBackPress = require('../screens/common/Pharmacy');  
    test('handleOnBackPress', () => {
        expect(handleOnBackPress()).toBeDefined();
    }); 
});
 


 