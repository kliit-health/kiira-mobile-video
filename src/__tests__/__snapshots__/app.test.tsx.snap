// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for reason type in appointment #195 appointment 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#0253E2",
        "height": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0253E2",
        "borderBottomColor": "#F9F9F9",
        "borderBottomWidth": 0,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Regular",
          "fontSize": 20,
          "fontWeight": "400",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "textAlign": "center",
        }
      }
    >
      Test Appointment
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#F6F7FA",
          "borderBottomWidth": 0.3,
          "borderTopColor": "#F6F7FA",
          "borderTopWidth": 0.3,
          "flexDirection": "row",
        },
        Object {
          "marginTop": 20,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="#8e8e93"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "fontFamily": "Poppins-Regular",
            "fontSize": 16,
            "paddingHorizontal": 15,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "margin": 10,
            "marginRight": 15,
          },
          undefined,
        ]
      }
    >
      <Search
        color="#8e8e93"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F9F9F9",
          "flex": 1,
          "flexDirection": "column",
          "width": 750,
        }
      }
    >
      <RCTScrollView
        data={
          Array [
            "2022-04-01",
            "2022-04-02",
            "2022-04-03",
            "2022-04-04",
            "2022-04-05",
            "2022-04-06",
            "2022-04-07",
            "2022-04-08",
            "2022-04-09",
            "2022-04-10",
            "2022-04-11",
            "2022-04-12",
            "2022-04-13",
            "2022-04-14",
            "2022-04-15",
            "2022-04-16",
            "2022-04-17",
            "2022-04-18",
            "2022-04-19",
            "2022-04-20",
            "2022-04-21",
            "2022-04-22",
            "2022-04-23",
            "2022-04-24",
            "2022-04-25",
            "2022-04-26",
            "2022-04-27",
            "2022-04-28",
            "2022-04-29",
            "2022-04-30",
            "2022-05-01",
            "2022-05-02",
            "2022-05-03",
            "2022-05-04",
            "2022-05-05",
            "2022-05-06",
            "2022-05-07",
            "2022-05-08",
            "2022-05-09",
            "2022-05-10",
          ]
        }
        decelerationRate="fast"
        extraData="2022-04-11"
        getItem={[Function]}
        getItemCount={[Function]}
        getItemLayout={[Function]}
        horizontal={true}
        initialScrollIndex={12}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onScrollToIndexFailed={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "column",
                  "height": 100,
                  "margin": 15,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                    },
                    Object {
                      "flexWrap": "wrap",
                    },
                  ]
                }
              >
                Apr
              </Text>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#008AFC",
                    "borderRadius": 50,
                    "borderWidth": 1,
                    "elevation": 3,
                    "marginVertical": 19.71,
                    "opacity": 1,
                    "padding": 8,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "elevation": 3,
                        "textAlign": "center",
                        "width": 18,
                      },
                      Object {
                        "flexWrap": "wrap",
                      },
                    ]
                  }
                >
                  1
                </Text>
              </View>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                    },
                    Object {
                      "flexWrap": "wrap",
                    },
                  ]
                }
              >
                Fri
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "width": 1560,
              }
            }
          />
        </View>
      </RCTScrollView>
      <Text
        style={
          Object {
            "color": "#000B1E",
            "fontFamily": "Poppins-Regular",
            "fontSize": 38,
            "height": 657,
            "margin": 20,
            "textAlign": "center",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 0,
      }
    }
  />
</View>
`;

exports[`Testing for reason type in appointment #195 expertTwillioLogin 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F9F9F9",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000B1E",
          "fontFamily": "Poppins-Regular",
          "fontSize": 20,
          "fontWeight": "400",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "textAlign": "center",
        }
      }
    >
      Kiira Video Chat
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "lightgrey",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "flex": 1,
            "marginTop": 10,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/logo-sm.png",
            }
          }
          style={
            Array [
              Object {
                "alignSelf": "center",
                "height": 150,
                "width": "90%",
              },
            ]
          }
        />
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderColor": "#008AFC",
              "borderRadius": 25,
              "borderWidth": 1,
              "elevation": 2,
              "height": 200,
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 13,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 10,
              "width": 637.5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Poppins-Bold",
                "fontSize": 16,
                "margin": 30,
                "textAlign": "center",
              }
            }
          >
            When you press Start your session will begin.
          </Text>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
              "margin": 20,
              "marginLeft": 50,
              "marginRight": 50,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#008AFC",
                "borderRadius": 20,
                "marginTop": 10,
                "opacity": 1,
                "paddingBottom": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
               START 
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="progressbar"
          accessibilityState={
            Object {
              "busy": false,
            }
          }
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "marginTop": 40,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 30,
                "opacity": 0,
                "width": 30,
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "height": 30,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 30,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 15,
                      "overflow": "hidden",
                      "width": 30,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "height": 30,
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "rotate": "-165deg",
                          },
                        ],
                        "width": 30,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 15,
                          "overflow": "hidden",
                          "width": 30,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        nativeID="animatedComponent"
                        style={
                          Object {
                            "borderColor": "#3f51b5",
                            "borderRadius": 15,
                            "borderWidth": 3,
                            "height": 30,
                            "width": 30,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "height": 30,
                    "transform": Array [
                      Object {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 30,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 15,
                      "overflow": "hidden",
                      "top": 15,
                      "width": 30,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    nativeID="animatedComponent"
                    style={
                      Object {
                        "height": 30,
                        "transform": Array [
                          Object {
                            "translateY": -15,
                          },
                          Object {
                            "rotate": "345deg",
                          },
                        ],
                        "width": 30,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 15,
                          "overflow": "hidden",
                          "width": 30,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        nativeID="animatedComponent"
                        style={
                          Object {
                            "borderColor": "#3f51b5",
                            "borderRadius": 15,
                            "borderWidth": 3,
                            "height": 30,
                            "width": 30,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 0,
      }
    }
  />
</View>
`;
